src/coreclr/tools/Common/Compiler/DisplayNameHelpers.cs

```diff
@@ -116,7 +117,58 @@ public static string GetDisplayName(this EventPseudoDesc @event)
 
         public static string GetDisplayName(this TypeDesc type)
         {
-            return Formatter.Instance.FormatName(type, FormatOptions.NamespaceQualify);
+            var sb = new StringBuilder();
+            if (type == null)
+                return sb.ToString();
+
+            Stack<TypeDesc> genericArguments = null;
+            switch (type)
+            {
+                // TODO: Linker handles here ArrayType and GenericInstanceType
+                default:
+                    string simpleName = Formatter.Instance.FormatName(type, FormatOptions.NamespaceQualify);
```

Comment:
Should we fix the formatter instead? Or is this how illink is doing it? (Trying to make sense of a string?)

I honestly wish we stopped trying to C#ify our names. Reflection doesn't do it either. It's fine.

Reply:
Reverted the changes in the DisplayNameHelpers in favor of the solution that is currently in the main branch that modifies the result checker instead.